src/programs/paymentManager/instruction.ts

```diff
@@ -0,0 +1,110 @@
+import { AnchorProvider, BN, Program } from "@project-serum/anchor";
+import type { Wallet } from "@saberhq/solana-contrib";
+import { TOKEN_PROGRAM_ID } from "@solana/spl-token";
+import type {
+  Connection,
+  PublicKey,
+  TransactionInstruction,
+} from "@solana/web3.js";
+import { SystemProgram } from "@solana/web3.js";
+
+import { CRANK_KEY } from "../tokenManager";
+import type { PAYMENT_MANAGER_PROGRAM } from ".";
+import { PAYMENT_MANAGER_ADDRESS, PAYMENT_MANAGER_IDL } from ".";
+import { findPaymentManagerAddress } from "./pda";
+
+export const init = async (
+  connection: Connection,
+  wallet: Wallet,
+  name: string,
+  params: {
+    makerFee: BN;
+    takerFee: BN;
+    feeScale: BN;
+    authority?: PublicKey;
+  }
+): Promise<[TransactionInstruction, PublicKey]> => {
+  const provider = new AnchorProvider(connection, wallet, {});
+
+  const paymentManagerProgram = new Program<PAYMENT_MANAGER_PROGRAM>(
+    PAYMENT_MANAGER_IDL,
+    PAYMENT_MANAGER_ADDRESS,
+    provider
+  );
+
+  const [paymentManagerId] = await findPaymentManagerAddress(name);
+
+  return [
+    paymentManagerProgram.instruction.init(
+      {
+        name: name,
+        makerFee: new BN(params.makerFee),
+        takerFee: new BN(params.takerFee),
+        feeScale: new BN(params.feeScale),
```

Comment:
already BNs looks like 

```suggestion
        makerFee: params.makerFee,
        takerFee: params.takerFee,
        feeScale: params.feeScale,
```

Reply:
resolved